biojava-core/src/main/java/org/biojava/nbio/core/search/io/blast/BlastXMLQuery.java

```diff
@@ -0,0 +1,270 @@
+package org.biojava.nbio.core.search.io.blast;
+
+
+import org.biojava.nbio.core.search.io.Hit;
+import org.biojava.nbio.core.search.io.Hsp;
+import org.biojava.nbio.core.search.io.Result;
+import org.biojava.nbio.core.search.io.ResultFactory;
+import java.io.File;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import javax.xml.xpath.XPathException;
+import org.biojava.nbio.core.sequence.template.Sequence;
+import org.biojava.nbio.core.util.XMLHelper;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+/**
+ * Re-designed by Paolo Pavan.
+ * You may want to find my contacts on Github and LinkedIn for code info 
+ * or discuss major changes.
+ * https://github.com/paolopavan
+ * 
+ *
+ * @author Scooter Willis <willishf at gmail dot com>
+ * @author Paolo Pavan
+ */
+public class BlastXMLQuery implements ResultFactory {
+    private static final Logger log = Logger.getLogger(BlastXMLQuery.class.getName());
+    Document blastDoc = null;
+    private File targetFile;
+    private List<Sequence> queryReferences, databaseReferences;
+    private Map<String,Sequence> queryReferencesMap, databaseReferencesMap;
+    
+    @Deprecated
+    public BlastXMLQuery(String blastFile) throws Exception {
+        readFile(blastFile);
+    }
+    
+    public BlastXMLQuery() {
+        
+    }
+    public void setFile(File f){
+        targetFile = f;
+    }
+    /**
+     * 
+     * @param maxEScore
+     * @return
+     * @throws Exception 
+     */
+    @Deprecated
+    public LinkedHashMap<String, ArrayList<String>> getHitsQueryDef(double maxEScore) throws Exception {
+        LinkedHashMap<String, ArrayList<String>> hitsHashMap = new LinkedHashMap<String, ArrayList<String>>();

```

Comment:
This actually escaped to me. Yes, the original class was
org.biojava.nbio.genome.query.BlastXMLQuery.
Since to an automatic usage search it seems to be poorly used, my original
will was to refactor this class moving from it's original position to
SearchIO while mantaining old method and declaring them obsolete.
Then caution reasons that you point out brought me to duplicate the class
and ask for advise to the mailing list.

My impression is that it is only a stub, to be further developed. If it is
so, we could just do as in my original plan because I have fear of
confusion it can generate since we are going to have new API for the task.
Otherwise I can clean my class as you are proposing.
What do you think?

2015-09-22 20:44 GMT+02:00 Jose Manuel Duarte <notifications@github.com>:

> In
> biojava-core/src/main/java/org/biojava/nbio/core/search/io/blast/BlastXMLQuery.java
> <https://github.com/biojava/biojava/pull/329#discussion_r40126844>:
>
> > +
> > +    public BlastXMLQuery() {
> > +
> > +    }
> > +    public void setFile(File f){
> > +        targetFile = f;
> > +    }
> > +    /**
> > +     *
> > +     * @param maxEScore
> > +     * @return
> > +     * @throws Exception
> > +     */
> > +    @Deprecated
> > +    public LinkedHashMap<String, ArrayList<String>> getHitsQueryDef(double maxEScore) throws Exception {
> > +        LinkedHashMap<String, ArrayList<String>> hitsHashMap = new LinkedHashMap<String, ArrayList<String>>();
>
> If I understand it correctly you have copied the
> org.biojava.nbio.genome.query.BlastXMLQuery into
> org.biojava.nbio.core.search.io.blast.BlastXMLQuery and then added all
> the new parsing stuff to it. That's fine, but as you kept the original one
> intact (which I think is a good idea), there's no reason to keep these
> deprecated methods around. You can simply remove them if they are not in
> use. We really don't want a new class to be born with deprecated methods
> because that doesn't make much sense. Deprecations are to support API
> transitions, not for new parts of the API.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biojava/biojava/pull/329/files#r40126844>.
>